Content.Client/Atmos/Overlays/AtmosDebugGraphOverlay.cs

```diff
@@ -0,0 +1,241 @@
+        {
+            ScreenSpaceDraw(args);
+        }
+    }
+
+    private void WorldSpaceDraw(in OverlayDrawArgs args)
+    {
+        var drawHandle = args.WorldHandle;
+
+        var mapId = args.Viewport.Eye!.Position.MapId;
+        var worldBounds = args.WorldBounds;
+
+        foreach (var mapGrid in _mapManager.FindGridsIntersecting(mapId, worldBounds))
+        {
+            if (!_atmosDebugOverlaySystem.HasData(mapGrid.Index))
+                continue;
+
+            drawHandle.SetTransform(mapGrid.WorldMatrix);
+
+            foreach (var tile in mapGrid.GetTilesIntersecting(worldBounds))
+            {
+                var dataMaybeNull = _atmosDebugOverlaySystem.GetData(mapGrid.Index, tile.GridIndices);
+                if (!dataMaybeNull.HasValue) continue;
+
+                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.BlockDirections) != 0)
+                    DrawBlockDirections(drawHandle, tile, dataMaybeNull.Value);
+                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.GasMoles) != 0)
+                    DrawGasMolesBarChart(drawHandle, tile, dataMaybeNull.Value);
+                if ((_atmosDebugOverlaySystem.CfgMode & AtmosDebugShowMode.FlowDirections) != 0)
+                    DrawFlowDirections(drawHandle, tile, dataMaybeNull.Value);
+            }
+        }
+
+        drawHandle.SetTransform(Matrix3.Identity);
+    }
+
+    private void DrawGasMolesBarChart(DrawingHandleWorld drawHandle,
+        TileRef tile,
+        SharedAtmosDebugOverlaySystem.AtmosDebugOverlayData data)
+    {
+        var topValueToShow = _atmosDebugOverlaySystem.CfgScale + _atmosDebugOverlaySystem.CfgBase;
+        for (var i = 0; i < Atmospherics.TotalNumberOfGases; i++)
+        {
+            if (data.Moles[i] > topValueToShow)
+            {
+                topValueToShow = data.Moles[i];
+            }
+        }
+
+        {
+            var lineZeroX = tile.X + 0.55f;
+            var linesBottomY = tile.Y + 0.15f;
+            const float linesMaxLen = 0.4f;
+
+            for (var i = 0; i < Atmospherics.TotalNumberOfGases; i++)
+            {
+                if (data.Moles[i] == 0f) continue;
+                var lineBottom = new Vector2(lineZeroX + (0.05f * i), linesBottomY);
+                var interp = (float)
+                    (Math.Log10(data.Moles[i] - _atmosDebugOverlaySystem.CfgBase) /
+                     Math.Log10(topValueToShow - _atmosDebugOverlaySystem.CfgBase));
+                if (interp < 0f) interp = 0.1f;
+                var lineTop = lineBottom + new Vector2(0, linesMaxLen * interp);
+                drawHandle.DrawLine(lineBottom, lineTop, Atmospherics.GasDisplayColors[i]);
+            }
+        }
+    }
+
+    private static ValueTuple<Vector2, Vector2> TileFaceInDirection(TileRef tile, AtmosDirection dir)
+    {
+        // Account for South being 0.
+        var atmosAngle = dir.ToAngle() - Angle.FromDegrees(90);
```

Comment:
is this supposed to be + or -90?

```suggestion
        var atmosAngle = dir.ToWorldAngle();
```

Reply:
also this seems duplicated with some stuff in `BlockOverlay`, is there a way to dedup this at all